cloud/services/subnets/subnets.go

```diff
@@ -58,100 +47,94 @@ func (s *Service) getExisting(ctx context.Context, rgName string, spec *Spec) (*
 }
 
 // Reconcile gets/creates/updates a subnet.
-func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
-	subnetSpec, ok := spec.(*Spec)
-	if !ok {
-		return errors.New("Invalid Subnet Specification")
-	}
-	existingSubnet, err := s.getExisting(ctx, s.Scope.Vnet().ResourceGroup, subnetSpec)
-	if err == nil {
-		// subnet already exists, update the spec and skip creation
-		var subnet *infrav1.SubnetSpec
-		if subnetSpec.Role == infrav1.SubnetControlPlane {
-			subnet = s.Scope.ControlPlaneSubnet()
-		} else if subnetSpec.Role == infrav1.SubnetNode {
-			subnet = s.Scope.NodeSubnet()
-		} else {
-			return nil
-		}
-
-		subnet.Role = subnetSpec.Role
-		subnet.Name = existingSubnet.Name
-		subnet.CidrBlock = existingSubnet.CidrBlock
-		subnet.ID = existingSubnet.ID
-
-		return nil
-	}
-	if !azure.ResourceNotFound(err) {
-		if err != nil {
-			return errors.Wrap(err, "failed to get subnet")
-		}
-	}
-	if !s.Scope.Vnet().IsManaged(s.Scope.ClusterName()) {
-		// if the vnet is unmanaged, we expect all subnets to be created as well
-		return fmt.Errorf("vnet was provided but subnet %s is missing", subnetSpec.Name)
-	}
+func (s *Service) Reconcile(ctx context.Context) error {
+	for _, subnetSpec := range s.Scope.SubnetSpecs() {
+		existingSubnet, err := s.getExisting(ctx, s.Scope.Vnet().ResourceGroup, subnetSpec)
+		switch {
+		case err != nil && !azure.ResourceNotFound(err):
+			return errors.Wrapf(err, "failed to get subnet %s", subnetSpec.Name)
+		case err == nil:
+			// subnet already exists, update the spec and skip creation
+			var subnet *infrav1.SubnetSpec
+			if subnetSpec.Role == infrav1.SubnetControlPlane {
+				subnet = s.Scope.ControlPlaneSubnet()
+			} else if subnetSpec.Role == infrav1.SubnetNode {
+				subnet = s.Scope.NodeSubnet()
+			} else {
+				return nil
```

Comment:
One more `return nil`. Not sure that we should what we should do here. Really, this should probably be a `continue` or return an error for an unexpected role. I lean toward ignore and continue.

Reply:
done